internal/app/transport/request.go

```diff
@@ -0,0 +1,123 @@
+package transport
+
+import (
+	discoveryv2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	structpb "github.com/golang/protobuf/ptypes/struct"
+	status "google.golang.org/genproto/googleapis/rpc/status"
+)
+
+// RequestVersion holds either one of the v2/v3 DiscoveryRequests
+type RequestVersion struct {
+	V2 *discoveryv2.DiscoveryRequest
+}
+
+// Request is the generic interface to abstract v2 and v3 DiscoveryRequest types
+type Request interface {
```

Comment:
I wonder if it makes sense to move this to go-control-plane. Doesn't have to be done now.

Reply:
Sure, we could try to remove the duplicated code in go-control-plane with something like this and move it there subsequently.